src/main/java/edu/oakland/arttour/dao/ArtTourDAO.java

```diff
@@ -15,20 +16,262 @@
+    jdbcTemplate.update(Constants.CREATE_TOUR, new Object[] {email, tourName});
+  }
+
+  public void deleteArtwork(String artworkId) {
+    jdbcTemplate.update(Constants.DELETE_ARTWORK, artworkId);
+  }
+
+  public void deleteCreator(int creatorId) {
+    jdbcTemplate.update(Constants.DELETE_CREATOR, creatorId);
+  }
+
+  public void deleteFavoriteArtwork(String email, String artworkId) {
+    jdbcTemplate.update(Constants.DELETE_FAVORITE_ARTWORK, new Object[] {email, artworkId});
+  }
+
+  public void deleteFavoriteCreator(String email, int creatorId) {
+    jdbcTemplate.update(Constants.DELETE_FAVORITE_CREATOR, new Object[] {email, creatorId});
+  }
+
+  public void deleteFavoriteTour(String email, int tourId) {
+    jdbcTemplate.update(Constants.DELETE_FAVORITE_TOUR, new Object[] {email, tourId});
+  }
+
+  public void deleteFromTour(int tourId, String artworkId) {
+    jdbcTemplate.update(Constants.DELETE_FROM_TOUR, new Object[] {tourId, artworkId});
+  }
+
+  public void deleteLocation(int locationId) {
+    jdbcTemplate.update(Constants.DELETE_LOCATION, locationId);
+  }
+
+  public void deleteTour(int tourId) {
+    jdbcTemplate.update(Constants.DELETE_TOUR, tourId);
+  }
+
+  public void deleteUser(String email) {
+    jdbcTemplate.update(Constants.DELETE_USER, email);
+  }
+
+  public void favoriteArtwork(String email, String artworkId) {
+    jdbcTemplate.update(Constants.FAVORITE_ARTWORK, new Object[] {email, artworkId});
+  }
+
+  public void favoriteCreator(String email, int creatorId) {
+    jdbcTemplate.update(Constants.FAVORITE_CREATOR, new Object[] {email, creatorId});
+  }
+
+  public void favoriteTour(String email, int tourId) {
+    jdbcTemplate.update(Constants.FAVORITE_TOUR, new Object[] {email, tourId});
+  }
+  
+  public List<Location> getAllLocations() {
+    return jdbcTemplate.query(Constants.GET_ALL_LOCATIONS, Location.mapper);
+  }
+
+  public Artwork getArtwork(String artworkId) throws DataAccessException {
+    return jdbcTemplate.queryForObject(Constants.GET_ARTWORK, Artwork.mapper, artworkId);
+  }
+
+  public List<String> getArtworkIds(int tourId) throws DataAccessException {
+    return jdbcTemplate.queryForList(Constants.GET_ARTWORK_IDS, String.class, tourId);
+  }
+
+  public List<Artwork> getCollection() throws DataAccessException {
+    return jdbcTemplate.query(Constants.GET_COLLECTION, Artwork.mapper);
+  }
+
+  public Creator getCreator(int creatorId) throws DataAccessException {
+    return jdbcTemplate.queryForObject(Constants.GET_CREATOR, Creator.mapper, creatorId);
+  }
+
+  public List<String> getFavoriteArtworkIds(String email) throws DataAccessException {
+    return jdbcTemplate.queryForList(Constants.GET_FAVORITE_ARTWORK_IDS, String.class, email);
+  }
+
+  public List<Integer> getFavoriteCreatorIds(String email) throws DataAccessException {
+    return jdbcTemplate.queryForList(Constants.GET_FAVORITE_CREATOR_IDS, Integer.class, email);
```

Comment:
Why is it integer class in here?

Reply:
I have to pass that for the `queryForList` method. I can't just pass sql and args

I struggled a lot with `jdbcTemplate` lol. and a bunch of their methods have changed

Reply:
Hmm interesting